infrastructure/policy/iam.tf

```diff
@@ -924,4 +924,58 @@ resource "aws_iam_role_policy" "authorize_lambda_invoke_policy" {
   ]
 }
 EOF
+}
+
+# ----------------------------------------------------------------------------------------------------------------------
+# AWS IAM COGNITO TRIGGER PRE SIGN UP
+# ----------------------------------------------------------------------------------------------------------------------
+
+resource "aws_iam_role" "verify_users_with_same_email_lambda_exec_role" {
+  name               = "${var.environment}_verify_users_with_same_email_exec_role"
+  path               = "/"
+  description        = "Allows Lambda Function to call AWS services on your behalf."
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "lambda.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy" "verify_users_with_same_email_lambda_list_users_policy" {
+  name        = "${var.environment}_verify_users_with_same_email_lambda_list_users_policy"
+  role        = aws_iam_role.verify_users_with_same_email_lambda_exec_role.id
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "cognito-idp:ListUsers",
+      "Effect": "Allow",
+      "Resource": "${var.cognito_user_pool_arn}"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy_attachment" "verify_users_with_same_email_lambda_logs" {
```

Comment:
This is only needed when you create the policy outside the role, please remove it.